scripts/errorCheck.ts

```diff
@@ -31,22 +30,18 @@ fs.readFile("src/compiler/diagnosticMessages.json", "utf-8", (err, data) => {
                     const msg = keys.filter(k => messages[k].code === errCode)[0];
                     messages[msg].seen = true;
                 }
-
-                done();
             });
-        }));
+        });
 
-        async.parallelLimit(tasks, 25, done => {
-            console.log("== List of errors not present in baselines ==");
-            let count = 0;
-            for (const k of keys) {
-                if (messages[k].seen !== true) {
-                    console.log(k);
-                    count++;
-                }
+        console.log("== List of errors not present in baselines ==");

```

Comment:
this code is executed synchronously before any of the files are actually read and processed, i.e. the callback of `readFile` is not yet called.

consider using `fs.promises` and `Promise.all` to wait for completion of all files or use the sync version

Reply:
Yes, that's true. I have another commit that switches this to proper async, but I hadn't yet pushed it.